Intermittent test failure: test_edit_outcome_summary_form_updates_experiment

Seen here: https://github.com/mozilla/experimenter/actions/runs/24410622844/job/71306088938?pr=15241

Looks like the random results test data is hitting an edge case:

```py
[4](https://github.com/mozilla/experimenter/actions/runs/24410622844/job/71306088938?pr=15241#step:5:6895)
=================================== FAILURES ===================================
_ EditOutcomeSummaryFormTests.test_edit_outcome_summary_form_updates_experiment _
[gw3] linux -- Python 3.13.6 /usr/local/bin/python3.13

self = <experimenter.nimbus_ui.tests.test_forms.EditOutcomeSummaryFormTests testMethod=test_edit_outcome_summary_form_updates_experiment>

    def test_edit_outcome_summary_form_updates_experiment(self):
>       experiment = NimbusExperimentFactory.create_with_lifecycle(
            NimbusExperimentFactory.Lifecycles.ENDING_APPROVE_APPROVE,
            takeaways_summary="",
            next_steps="",
            project_impact="",
        )

experimenter/nimbus_ui/tests/test_forms.py:5657: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
experimenter/experiments/tests/factories.py:1125: in create_with_lifecycle
    experiment.monitoring_data = build_random_monitoring_data(
experimenter/experiments/tests/factories.py:477: in build_random_monitoring_data
    top_reason: {"1pct_count": random.randint(50, unenrollments or 100)},
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/local/lib/python3.13/random.py:340: in randint
    return self.randrange(a, b+1)
           ^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <random.Random object at 0x564f880c24d0>, start = 50, stop = 48, step = 1

    def randrange(self, start, stop=None, step=_ONE):
        """Choose a random item from range(stop) or range(start, stop[, step]).
    
        Roughly equivalent to ``choice(range(start, stop, step))`` but
        supports arbitrarily large ranges and is optimized for common cases.
    
        """
    
        # This code is a bit messy to make it fast for the
        # common case while still doing adequate error checking.
        istart = _index(start)
        if stop is None:
            # We don't check for "step != 1" because it hasn't been
            # type checked and converted to an integer yet.
            if step is not _ONE:
                raise TypeError("Missing a non-None stop argument")
            if istart > 0:
                return self._randbelow(istart)
            raise ValueError("empty range for randrange()")
    
        # Stop argument supplied.
        istop = _index(stop)
        width = istop - istart
        istep = _index(step)
        # Fast path.
        if istep == 1:
            if width > 0:
                return istart + self._randbelow(width)
>           raise ValueError(f"empty range in randrange({start}, {stop})")
E           ValueError: empty range in randrange(50, 48)

/usr/local/lib/python3.13/random.py:323: ValueError
================================ tests coverage ================================
```

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-6780)

--- a/experimenter/experimenter/experiments/tests/factories.py
+++ b/experimenter/experimenter/experiments/tests/factories.py
@@ -474,7 +474,7 @@ def build_random_monitoring_data(branches):
         }
         top_reason = random.choice(UNENROLLMENT_REASONS)
         reasons_by_branch[name] = {
-            top_reason: {"1pct_count": random.randint(50, unenrollments or 100)},
+            top_reason: {"1pct_count": random.randint(50, max(unenrollments, 100))},
         }
 
     total_unenrollments = sum(b["unenrollments"] for b in branch_data.values())

